src/DonationForms/Properties/FormSettings.php

```diff
@@ -243,6 +243,12 @@ public static function fromArray(array $array): self
         $self->multiStepNextButtonText = $array['multiStepNextButtonText'] ?? __('Continue', 'give');
         $self->pdfSettings = isset($array['pdfSettings']) && is_array($array['pdfSettings']) ? $array['pdfSettings'] : [];
 
+        $donationFormId = $_GET['donationFormID'] ?? false;
+        if (count($self->pdfSettings) === 0 && $donationFormId) {
+            $self->pdfSettings = apply_filters('givewp_form_builder_pdf_settings_default_values', [],
+                (int)$donationFormId);
+        }
```

Comment:
@glaubersilva we don't want to use the `$_GET` super global here as the intention of this class is a model property so it shouldn't be interacting with the request directly.  The form ID should be found upstream where this property lives.

I'm also a little confused why we need a filter here, can you explain and we can see if this is the right solution?

Reply:
Yeah, makes sense! Resolved in bed3b3f7a6df48ff98939d2c7a4cacd19dc409ab

About the filter, we need to use it to be able to set the default values on the PDF Receipt addon side. 

We are using this filter here:

https://github.com/impress-org/give-pdf-receipts/pull/253/files#diff-51e6882460954791c9102a76f0aa70e20e702984407d70dd3c962e8fb3c01414

Reply:
Hmm, is this just an alternative for a form migration step?  When I brought up the idea of migrating all existing meta - that was intended as a backup for unknown data.  If we know the data that needs to be migrated then lets go ahead and add a migration step.  

As far as this filter goes, i'm not completely convinced it's necessary.  If you need default values for the form builder settings wouldn't you just add those through the localized object in the add-on?  Happy to chat through this one as well.

Reply:
Also, in case you didn't know about these - we have two very useful actions in which you can update the form model either during render or creation:

`givewp_donation_form_schema` and `givewp_form_builder_new_form`

Reply:
@jonwaldstein Gotcha your point about the filter, so I removed it and added the default settings to the localized object in the addon. Resolved in 8cdb6323daa2d89032d591c075738fbbcce723c0

About your question:

> Is this just an alternative for a form migration step?

Not necessarily, as these meta keys still are being updated on the addon side (when the form builder gets updated) to keep backward compatibility, so the new forms still need to have these meta keys. Reference: https://github.com/impress-org/give-pdf-receipts/blob/develop/src/FormExtension/FormBuilder/Actions/UpdatePdfSettingsMeta.php

> If we know the data that needs to be migrated then lets go ahead and add a migration step that includes updating the form settings object.

Initially, I also thought this way, but due to the reason I mentioned above I changed my mind. Another reason that I thought would be nice to handle it on the addon side is to be able to use the DTO class that lives there to provide better context about the data which are being handled.

With that in mind, do you think we still need a new step exclusive to the pdf settings? 

If yes, maybe we wouldn't even need that other PR:  https://github.com/impress-org/give-pdf-receipts/pull/253

Reply:
As discussed in a call, we decided to close another PR related to this one and I created a new migration step for the Pdf Settings here on this PR.